src/native/libs/System.Security.Cryptography.Native.Apple/pal_networkframework.m

```diff
@@ -0,0 +1,644 @@
+    // 
+    // The trick works by layering a custom framer on top of this dummy connection,
+    // then adding TLS on top of the framer. The framer intercepts the raw TLS data
+    // and exposes it to SslStream, preventing it from ever reaching the underlying
+    // connection.
+    //
+    // The endpoint values (127.0.0.1:42) are arbitrary - they just need to be
+    // syntactically and semantically valid since the connection is never established.
+    //
+    // TODO: reuse parameters and endpoint across connections
+    nw_parameters_t parameters = nw_parameters_create_secure_udp(NW_PARAMETERS_DISABLE_PROTOCOL, NW_PARAMETERS_DEFAULT_CONFIGURATION);
+    nw_endpoint_t endpoint = nw_endpoint_create_host("127.0.0.1", "42");
+
+    nw_connection_t connection = nw_connection_create(endpoint, parameters);
+
+    // connection retains its own reference to the endpoint and parameters
+    nw_release(endpoint);
+    nw_release(parameters);
+
+    return connection;
+}
+
+// This writes encrypted TLS frames to the safe handle. It is executed on NW Thread pool
+static nw_framer_output_handler_t framer_output_handler = ^(nw_framer_t framer, nw_framer_message_t message, size_t message_length, bool is_complete)
+{
+
+    if (__builtin_available(macOS 12.3, iOS 15.4, tvOS 15.4, watchOS 2.0, *))
+    {
+        nw_protocol_options_t framer_options = nw_framer_copy_options(framer);
+
+        NSNumber* num = nw_framer_options_copy_object_value(framer_options, "GCHANDLE");
+        assert(num != NULL);
+
+        void * ptr;
+        [num getValue:&ptr];
+        [num release];  // Release the NSNumber after extracting the value
+        size_t size = message_length;
+
+        nw_framer_parse_output(framer, 1, message_length, NULL, ^size_t(uint8_t *buffer, size_t buffer_length, bool is_complete2) {
+            void* length = (void*)buffer_length;
+            (_writeFunc)(ptr, buffer, &length);
+            (void)is_complete2;
+            (void)message;
+            return buffer_length;
+        });
+
+        nw_release(framer_options);
+    }
+    else
+    {
+        assert(0);
+    }
+    (void)is_complete;
+};
+
+static nw_framer_stop_handler_t framer_stop_handler = ^bool(nw_framer_t framer) {
+    if (__builtin_available(macOS 12.3, iOS 15.4, tvOS 15.4, watchOS 8.4, *))
+    {
+        size_t state = 0;
+        nw_protocol_options_t framer_options = nw_framer_copy_options(framer);
+        NSNumber* num = nw_framer_options_copy_object_value(framer_options, "GCHANDLE");
+        if (num != NULL)
+        {
+            [num getValue:&state];
+            [num release];  // Release the NSNumber after extracting the value
+            (_statusFunc)(state, PAL_NwStatusUpdates_FramerStop, 0, 0, NULL);
+        }
+
+        nw_release(framer_options);
+    }
+
+    return TRUE;
+};
+
+static nw_framer_cleanup_handler_t framer_cleanup_handler = ^(nw_framer_t framer) {
```

Comment:
nit: consider placing `{` on its own line or same line consistently in this file.